### PR TITLE
8th November 2021 | Yet more dependency updates

### PR DESCRIPTION
**Pull request by coldlink**
## What does this change?
- Updates dependencies to the latest versions where possible

## Notes
- ~`react-router-dom` is a major update, so will move into a separate PR~
  - Updated `react-router-dom` in this PR and fix all breaking changes using this [upgrade guide](https://reactrouter.com/docs/en/v6/upgrading/v5#upgrade-to-react-router-v6)
  - Had to refactor how we handle `404`/not found routes as react router wasn't picking it up otherwise and causing a `502` error on the server

## Tested?
- [x] Local
- [x] CI
- [ ] CODE

### CODE REVIEW

File: src/client/components/PasswordInput.tsx [hunk 1]
```diff
@@ -29,7 +33,7 @@ export const isDisplayEyeOnBrowser = (browserName: string | undefined) => {
 
 const textInputBorderStyle = (error?: string) => css`
   border: ${error
-    ? `4px solid ${textInputDefault.textInput.textError}`
+    ? `4px solid ${errorColors[400]}`
```
Comment (ob6160) [inline src/client/components/PasswordInput.tsx:36]: I think think one can be `textInput.borderError` maybe

**Review by ob6160 — COMMENTED** [review]
lgtm, one small comment on colour token

File: src/server/routes/index.ts [hunk 2]
```diff
@@ -47,8 +48,10 @@ uncachedRoutes.use(verifyEmail);
 uncachedRoutes.use(magicLink);
 
 // welcome routes
-uncachedRoutes.use(noCache, welcome);
+uncachedRoutes.use(welcome);
 
 router.use(uncachedRoutes);
 
+router.use(notFound);
```
Comment (QuarpT) [inline src/server/routes/index.ts:55]: if res.locals gets passed to 404 then this should be uncached to avoid leaking sensitive data 🤔

**Review by QuarpT — DISMISSED** [review]
worthing looking at this since it could be a potential vulnerability
https://github.com/guardian/gateway/pull/1112/files?diff=split&w=1#r744637606

File: src/client/routes.tsx [hunk 3]
```diff
@@ -34,101 +34,106 @@ export type RoutingConfig = {
...
-    </Route>
-    <Route exact path={`${Routes.WELCOME}${Routes.EMAIL_SENT}`}>
-      <EmailSentPage />
-    </Route>
-    <Route exact path={`${Routes.WELCOME}${Routes.TOKEN_PARAM}`}>
-      <WelcomePage />
-    </Route>
+  <RouterRoutes>
+    <Route path={Routes.SIGN_IN} element={<SignInPage />} />
+    <Route path={Routes.REGISTRATION} element={<RegistrationPage />} />
```
Comment (QuarpT) [inline src/client/routes.tsx:39]: the `exact` prop has been removed from some of these. Does this break anything?

Comment (coldlink) [reply src/client/routes.tsx:39]: From the upgrade guide https://reactrouter.com/docs/en/v6/upgrading/v5#upgrade-to-react-router-v6:

<Route exact> is gone. Instead, routes with descendant routes (defined in other components) use a trailing * in their path to indicate they match deeply

Comment (coldlink) [reply src/client/routes.tsx:39]: So from that reasoning every route is "`exact`" by default, and you use a `*` if it's not exact.

In gateway every route is exact, so it won't break anything now by removing it.

File: src/server/routes/index.ts (hunk 2 above)
Comment (coldlink) [reply src/server/routes/index.ts]: Good spot!

File: src/client/routes.tsx (hunk 3 above)
Comment (QuarpT) [reply src/client/routes.tsx:39]: 👍

**Review by ob6160 — APPROVED** [review]

**Pull request merged**